src/modules/internet/user-agent.ts

```diff
@@ -179,39 +155,29 @@ export function generate(faker: Faker): string {
     ]);
 
   const randomBrowserAndOS = (): [Browser, OS] => {
-    const browser: Browser = weightedKeyFromObject({
-      chrome: 0.45132810566,
-      iexplorer: 0.27477061836,
-      firefox: 0.19384170608,
-      safari: 0.06186781118,
-      opera: 0.01574236955,
-    });
-    const os: OS = weightedKeyFromObject(
-      {
-        chrome: { win: 0.89, mac: 0.09, lin: 0.02 },
-        firefox: { win: 0.83, mac: 0.16, lin: 0.01 },
-        opera: { win: 0.91, mac: 0.03, lin: 0.06 },
-        safari: { win: 0.04, mac: 0.96 },
-        iexplorer: { win: 1 },
-      }[browser]
-    );
+    const browserToOsMap = {
+      chrome: ['win', 'mac', 'lin'],
+      firefox: ['win', 'mac', 'lin'],
+      opera: ['win', 'mac', 'lin'],
+      safari: ['win', 'mac'],
+      iexplorer: ['win'],
+    } satisfies Record<Browser, OS[]>;
+    const browser: Browser = faker.helpers.objectKey(browserToOsMap);
+    const os: OS = faker.helpers.arrayElement(browserToOsMap[browser]);
```

Comment:
These can be auto-inferred and are checked at the function's return boundary

```suggestion
    const browser = faker.helpers.objectKey(browserToOsMap);
    const os = faker.helpers.arrayElement(browserToOsMap[browser]);
```

Reply:
yes, it can auto-infer those types, but what if I made an error in the typing of `browserToOsMap` (for example forgot the `satisfied`). Suddenly my `browser` variable becomes `string`. I explicitly lef them in to ensure that I'm working with a `Brwoser` and `OS` type at this point (and if not TypeScript should tell me immediately and not laten when `string is not assignable to Browser`).

Reply:
I could also search for a lint rule for that, so take it as not required for this PR as needed

Reply:
I think you can take a look at [@typescript-eslint/no-inferrable-types](https://typescript-eslint.io/rules/no-inferrable-types/) if you want to enforce this 👍

Reply:
It's already configured :eyes: 
https://github.com/faker-js/faker/blob/6772b00cf19ccdbebefe023dc0f96c6576234b5b/.eslintrc.js#L46-L49

So if this does not complain, I don't complain either :shrug: